src/main/recorder.ts

```diff
@@ -0,0 +1,181 @@
+import { Metadata } from './logutils';
+import { writeMetadataFile, runSizeMonitor, getNewestVideo, deleteVideo, cutVideo } from './util';
+import { mainWindow }  from './main';
+import { app } from 'electron';
+import path from 'path';
+
+const obsRecorder = require('./obsRecorder');
+const fs = require('fs');
+const glob = require('glob');
+
+/**
+ * Represents an OBS recorder object.
+ */
+ class Recorder {
+    private _isRecording: boolean = false;
+    private _isRecordingBuffer: boolean = false;
+    private _storageDir: string;
+    private _tempStorageDir: any;
+    private _maxStorage: number;
+    private _bufferIntervalID?: any;
+
+    /**
+     * Constructs a new Recorder.
+     */
+    constructor(storageDir: string, maxStorage: number) {
+        this._storageDir = storageDir;
+        this._maxStorage = maxStorage;       
+        this._tempStorageDir = path.join(app.getPath("temp"), "WarcraftRecorder"); // C:\Users\alexa\AppData\Local\Temp\WarcraftRecorder
+    }
+
+    /**
+     * 
+     */
+     get isRecording() {
+        return this._isRecording;
+    }
+
+    /**
+     * 
+     */
+    set isRecording(value) {
+        this._isRecording = value;
+    }
+
+    /**
+     * 
+     */
+    get isRecordingBuffer() {
+        return this._isRecordingBuffer;
+    }
+    
+    /**
+     * 
+     */
+    set isRecordingBuffer(value) {
+        this._isRecordingBuffer = value;
+    }
+
+    /**
+     * init
+     */
+    init = () => {
```

Comment:
is there any reason we really need this init to be called? init methods are a little awkward for the consumers to use, most of the time this type of stuff can just be done in the constructor. 

if you prefer to have this, do we want any protection around this -- IE illegal state exceptions if this hasnt been called first

Reply:
no reason- i'll add to the constructor